### DISCUSSION TITLE
I Need Help Submitting a pull request for Homebrew Gdal

### DISCUSSION THREAD

**Discussion by OldSubSailor** [Homebrew/discussions#1928] [Getting started]
I have been told many times to submit a pull request for various concerns, so I took the little Github tutorial and things seemed straight forward enough. 
The tutorial process that I went through was to click on the main branch pull down and create a new branch and from that branch one proceeds to ultimately create a pull request. I thought that I understood the basics. 
Unfortunately there seems to be a general assumption that just because a person stumbles upon Github, one knows all the details of using it. Hence, one often receives rather terse, unhelpful responses. 

When I went to Homebrew Github I found 20,925 repository results which is a massive confusion factor. I tried to narrow it down by searching for Homebrew Gdal, which returns 5 repository results; the latest repository being empty the others either frozen or several years old, to be honest I do not know if any is the one that is downloaded via brew install.  Brew Info leads me to think not. 

I get this information from the the terminal command line, perhaps someone can help me with it and tell me where I can actually find the GDAL repository.

% brew info GDAL
gdal: stable 3.3.1 (bottled), HEAD
Geospatial Data Abstraction Library
https://www.gdal.org/
Conflicts with:
  avce00 (because both install a cpl_conv.h header)
  cpl (because both install cpl_error.h)
/usr/local/Cellar/gdal/3.3.1_2 (356 files, 58.9MB) *
  Poured from bottle on 2021-07-30 at 15:25:20
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb
License: MIT
==> Dependencies
Build: pkg-config ✔
Required: . . ., proj@7 ✔, . . .
There is an apparent issue with proj@7, a warning is generated when Gdal is installed via Homebrew and I have discussed this before. 
I am trying to look through Github homebrew core. One thing I have found that I can't do is use the main branch dropdown to create a new branch. 
Your help and advice is appreciated. I am very new to all this.

**Comment by alebcay** [comment]
For any other maintainers arriving here, this seems to follow up from core issue https://github.com/Homebrew/homebrew-core/issues/82441.

> perhaps someone can help me with it and tell me where I can actually find the GDAL repository.

I'm not sure what you're asking here, but
- if you're looking for the GDAL formula (the instructions that tells Homebrew how to build/package GDAL), the file is at https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb.
- if you're looking for the GDAL repository (where GDAL maintainers actually build and maintain the software GDAL itself), the repository is at https://github.com/OSGeo/GDAL

> One thing I have found that I can't do is use the main branch dropdown to create a new branch.

Yes, only Homebrew team maintainers can create branches/modify branches directly on the Homebrew repository (https://github.com/Homebrew/homebrew-core).

You're going to want to [fork the Homebrew repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo) and then create the new branch and make whatever changes you wish to make.

**Comment by OldSubSailor** [comment]
Thank you for your reply. I appreciate that you took the time to help and educate me.

I forgot that I left this statement regarding finding GDAL. After stumbling around I finally found it after seeing the note stating that not all the files were shown in the core page. A search worked. Face red here. 

I submitted an issue and although I was evidently in error there also, another person has been kind enough to help me there and slowly teaching me about GDAL and Python changes and the api note regarding import gdal not working. 

A key learning point for me was that all these warnings are not bugs but evidently for information. Unfortunately when I asked about them in the discussions all that I was told was to generate a pull request. I really appreciate that Michael Cho has been so kind in responding to the Issue that I created in error. 

So the question does remain, even when I do try reading around in Homebrew and Github much of it is a mystery to me. I think that I am really not trying to make changes as much as trying to learn something  by asking about a concern that I have seen, but how and where to do it.   Do I really need a pull request for that?

Thank you again.
